### PR TITLE
[CAT-799] Timeout Parameter Not Set Up as Expected

### PR DESCRIPTION
**Pull request by nateshim-indico**
## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

## Checklist:

- [x] Jira story/task is put into PR title
- [x] Code has been self-reviewed
- [x] Hard-to-understand areas have been commented
- [x] Documentation has been updated
- [x] My changes generate no new warnings
- [x] Unit tests have been added for base functionality and known edge cases
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: indico/types/utils.py [hunk 1]
```diff
@@ -15,3 +16,16 @@ def _camel(match):
...
+    def __init__(self, timeout: int):
+        self.timeout = timeout
+
+    def _handler(self, signum, frame):
+        raise IndicoTimeoutError(self.timeout)
+
+    def run(self):
+        if self.timeout == 0:
+            raise IndicoTimeoutError(self.timeout)
+        signal.signal(signal.SIGALRM, self._handler)
```
Comment (meghanhickey) [inline indico/types/utils.py:30]: oh this is really slick! I dig it.

Reading the docs for the signal library, it looks like `signal.SIGALARM` is only available on Unix. I think we want this SDK to be usable on Windows as well-- is there a different signal you could use?

Comment (nateshim-indico) [reply indico/types/utils.py:30]: ah yep will do!

Comment (nateshim-indico) [reply indico/types/utils.py]: Looks like `signal` is just not overall a good choice as it's not Windows friendly in general. Just parsed out current timing logic into `Timer` and fixed the double time issues

**Comment by meghanhickey** [comment]
nice! this looks good to me, but I'd hold off on merging until @goatrocks takes a look

File: indico/queries/jobs.py [hunk 2]
```diff
@@ -53,7 +52,7 @@ class JobStatus(RequestChain):
     Args:
         id (int): id of the job to query for status.
         wait (bool, optional): Wait for the job to complete? Default is True
-        timeout (float or int, optional): Timeout after this many seconds.
```
Comment (goatrocks) [inline indico/queries/jobs.py:56]: Nice work on the fix + updating the tests!

Only feedback here - there's an impact to changing the interface/accepted types -- while you could submit an int for a float, if users are submitting floats and upgrade after we narrow the accepted types to int, they will have to change their scripts. 

If it's possible to accept floats here we should still. Thank you!

**Review by goatrocks — CHANGES_REQUESTED** [review]

**Review by goatrocks — APPROVED** [review]
Nice! Thank you!
